src/main/java/com/palantir/docker/compose/connection/waiting/ServiceWait.java

```diff
@@ -38,13 +40,30 @@ public ServiceWait(Container service, HealthCheck healthCheck, Duration timeout)
 
     public void waitTillServiceIsUp() {
         log.debug("Waiting for service '{}'", service);
+        final AtomicReference<Optional<SuccessOrFailure>> lastSuccessOrFailure = new AtomicReference<>(Optional.empty());
         try {
             Awaitility.await()
                     .pollInterval(50, TimeUnit.MILLISECONDS)
```

Comment:
Could you extract the `Consumer` into a static method or something? so the await would read like:

```java
...
.until(weHaveSuccess(lastSucessOrFailure))
```